hazelcast/src/main/java/com/hazelcast/spi/impl/proxyservice/impl/DistributedObjectEventPacket.java

```diff
@@ -60,12 +64,17 @@ public UUID getSource() {
         return source;
     }
 
+    public TenantControl getTenantControl() {
+        return tenantControl;
+    }
+
     @Override
     public void writeData(ObjectDataOutput out) throws IOException {
         out.writeBoolean(eventType == EventType.CREATED);
         out.writeUTF(serviceName);
         out.writeUTF(name);
         UUIDSerializationUtil.writeUUID(out, source);
+        out.writeObject(tenantControl);
```

Comment:
rolling upgrades: make this class implement `Versioned` and guard for cluster version being greater or equal than current one

(also in `readData` below)